povm_toolbox/library/classical_shadows.py

```diff
@@ -61,3 +65,15 @@ def __init__(
             shot_batch_size=shot_batch_size,
             seed_rng=seed_rng,
         )
+
+    @property
+    def kwargs(self) -> dict[str, Any]:
+        """Return the attributes of ``self`` needed to build a copy of ``self``."""
+        kwargs = {
+            "n_qubit": self.n_qubit,
+            "measurement_twirl": self.measurement_twirl,
+            "measurement_layout": self.measurement_layout,
+            "shot_batch_size": self.shot_batch_size,
+            "seed_rng": self._rng.bit_generator.state,
+        }
+        return kwargs
```

Comment:
This seems like an approach that can easily break down the line if more kwargs get added to the `__init__` and not here. Furthermore, having to overwrite this method for each subclass with different kwargs is a bit cumbersome.

Reply:
Also, not having any proper types is unfortunate. This would suggest another `dataclass` to be resolved which begs the question of what we really need to track.